### PR TITLE
Consistency and quality updates for java quickstart inner loop

### PR DESCRIPTION
**Pull request by paulyuk**
## Purpose
Makes a few changes to align with other quickstarts:
- breaks out `httpget` and `httppost` functions
- allows null name param in http get
- adds test.http file
- updates readmes and code
- fixes curl command
- upgrades to latest Functions SDK for Java
- removes Visual Studio steps (but emphasizes VS Code)
- optimized for JDK 17 +

### CODE REVIEW

File: local.settings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (ggailey777) [inline local.settings.json:1]: Should we be putting the `local.settings.json` file in the repo, which generally isn't a best practice? 
It's pretty easy to regenerate it locally using `func init`.

**Review by BigMorty — APPROVED** [review]
Looks good overall. I did not review the actual code addition for the post.
As Glenn mentioned, we should not commit the local.settings.json file.

File: local.settings.json (hunk 1 above)
Comment (paulyuk) [reply local.settings.json:1]: for now I vote no.  it's too easy for tools and users to put something private in there and check it in.

File: http/pom.xml [hunk 2]
```diff
@@ -13,7 +13,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <java.version>17</java.version>
         <azure.functions.maven.plugin.version>1.36.0</azure.functions.maven.plugin.version>
-        <azure.functions.java.library.version>2.2.0</azure.functions.java.library.version>
+        <azure.functions.java.library.version>3.1.0</azure.functions.java.library.version>
```
Comment (raorugan) [inline http/pom.xml:16]: core tools is generating 1.24 functions java library pom.xml. core tool templates need an update. is this tested with 3.1.0 version?

**Review by raorugan — COMMENTED** [review]
Left few comments! Can test this code post merge.

File: http/pom.xml (hunk 2 above)
Comment (paulyuk) [reply http/pom.xml:16]: This sample itself is.  I think we need to do the right thing despite what the templates are doing.  We should not be shipping 2.2.0 in templates!  Those are from 2022.  3.1 has been out there for 9 months.  Let's use it.
https://central.sonatype.com/artifact/com.microsoft.azure.functions/azure-functions-java-library/versions

**Comment by paulyuk** [comment]
Ready to merge @raorugan ?

File: http/pom.xml (hunk 2 above)
Comment (raorugan) [reply http/pom.xml:16]: Yes, I agree maven repo has 3.1 as well - [Maven Repository: com.microsoft.azure.functions » azure-functions-java-library (mvnrepository.com)](https://mvnrepository.com/artifact/com.microsoft.azure.functions/azure-functions-java-library)

LGTM !

**Pull request merged**